### PR TITLE
improve fixconfig

### PR DESCRIPTION
**Pull request by BenTheElder**
- fallback to naive file copy if rename doesn't work
- cleanup temp file
- remove cache disk related things from security jobs
- remove cache disk from test-infra jobs that aren't using it
- add `hack/update-config.sh` and `hack/verify-config.sh`
- add job to verify prow config
- remove `replace(..)` mess from `prow/config/config_test.go`, this means the test will only need to load the config once :tada:
- run updated fixconfig

### CODE REVIEW

**Comment by BenTheElder** [comment]
@krzyzacy this will fail the existing presubmit, WDYT about removing the security repo presubmit in favor a verify style check that ensures you've run `bazel run  //maintenance/fixconfig:fixconfig -- --config=$PWD/prow/config.yaml && bazel run //jobs:config_sort` ?

**Comment by krzyzacy** [comment]
how about run the generation in config_test and make sure things are still the same?

**Comment by BenTheElder** [comment]
Need to fix some pylint and add testgrid entries. Pretty done otherwise.

File: hack/update-config.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+TESTINFRA_ROOT=$(git rev-parse --show-toplevel)
+
+bazel run  //maintenance/fixconfig:fixconfig -- --config=${TESTINFRA_ROOT}/prow/config.yaml && \
```
Comment (krzyzacy) [inline hack/update-config.sh]: bikeshed: one less space between `run` and `//target`

File: maintenance/fixconfig/main.go [hunk 2]
```diff
@@ -121,6 +124,45 @@ func convertJobToSecurityJob(j *config.Presubmit) {
...
+		filteredVolumes := []kube.Volume{}
+		removedVolumeNames := sets.String{}
+		for _, volume := range j.Spec.Volumes {
+			if volume.VolumeSource.HostPath != nil &&
+				strings.HasPrefix(volume.VolumeSource.HostPath.Path, "/mnt/disks/ssd0") {
+				removedVolumeNames.Insert(volume.Name)
+				continue
+			}
+			filteredVolumes = append(filteredVolumes, volume)
+		}
```
Comment (krzyzacy) [inline maintenance/fixconfig/main.go]: assume you need to set `j.Spec.Volumes = filteredVolumes`

File: maintenance/fixconfig/main.go [hunk 3]
```diff
@@ -207,10 +250,25 @@ func main() {
...
-	// copy file to replace original
+	// move file to replace original
 	f.Close()
 	err = os.Rename(f.Name(), *configPath)
 	if err != nil {
-		log.Fatalf("Failed to replace config with updated version: %v", err)
+		// fallback to copying the file instead
+		src, err := os.Open(f.Name())
+		if err != nil {
+			log.Fatalf("Failed to replace config with updated version: %v", err)
```
Comment (krzyzacy) [inline maintenance/fixconfig/main.go]: use a different error message? all three fatalf here are the same

File: maintenance/fixconfig/main.go (hunk 2 above)
Comment (BenTheElder) [reply maintenance/fixconfig/main.go]: done, thanks

File: hack/update-config.sh (hunk 1 above)
Comment (BenTheElder) [reply hack/update-config.sh]: done

File: maintenance/fixconfig/main.go (hunk 3 above)
Comment (BenTheElder) [reply maintenance/fixconfig/main.go]: :shrug: different errors at the end of the message, but this message is the relevant part to the user. I should just move this to a function.

Comment (BenTheElder) [reply maintenance/fixconfig/main.go]: refactored

**Comment by BenTheElder** [comment]
fixed nits and rebased against latest config

**Comment by BenTheElder** [comment]
@krzyzacy finally fully rebased and green ✅ missed a testgrid change earlier from #6317.

File: prow/config.yaml [hunk 4]
```diff
@@ -6001,29 +5562,61 @@ presubmits:
       containers:
       - image: gcr.io/k8s-testimages/gcloud-in-go:v20171113-192bec25
         args:
+        - "--job=$(JOB_NAME)"
         - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
         - "--root=/go/src"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
-        - "--clean"
+        - --service-account=/etc/service-account/service-account.json
```
Comment (krzyzacy) [inline prow/config.yaml]: is this intended?

**Comment by BenTheElder** [comment]
/hold
discussed a different option for the cache volumes offline, need to be more intelligent for DinD at least and mount an emptyDir instead.

File: prow/config.yaml (hunk 4 above)
Comment (BenTheElder) [reply prow/config.yaml]: --clean was not though shouldn't hurt since no cache volume now. I'll rever that.

**Comment by BenTheElder** [comment]
/hold cancel
this handles dind jobs correctly now by always stripping /docker-graph mounts (and associated volumes) for DIND enabled jobs (detected by the env we require) and then adding an auto generated emptyDir + mount

**Pull request merged**